lib/galaxy/tool_util/deps/container_resolvers/mulled.py

```diff
@@ -322,7 +323,24 @@ def first_tag_if_available(image_name):
     return name
 
 
-class CachedMulledDockerContainerResolver(ContainerResolver):
+class DockerContainerResolver(ContainerResolver):
+
+    def __init__(self, *args, **kwargs):
+        self._docker_cli_available = bool(which('docker'))
+        super().__init__(*args, **kwargs)
+
+    @property
+    def docker_cli_available(self):
+        return self._docker_cli_available
+
+    @docker_cli_available.setter
+    def docker_cli_available(self, value):
+        if not value:
+            log.info('Docker CLI not available, cannot list or pull images in Galaxy process. Does not impact kubernetes.')
+        self._docker_cli_available = value
+
+
+class CachedMulledDockerContainerResolver(DockerContainerResolver):
 
     resolver_type = "cached_mulled"
     container_type = "docker"
```

Comment:
You could move this line to `DockerContainerResolver` and remove from all its subclasses.